File: java/Factorial.java
public class Factorial {
    public static void main(String[] args) {
        int n = 5;
        System.out.println(factorial(n));
    }

    public static int factorial(int n) {
        if (n == 0) {
            return 1;
        }
        return n * factorial(n - 1);
    }
}

File: csharp/Factorial.cs
using System;

class Factorial {
    static void Main(string[] args) {
        int n = 5;
        Console.WriteLine(Factorial(n));
    }

    static int Factorial(int n) {
        if (n == 0) return 1;
        return n * Factorial(n - 1);
    }
}
